app/javascript/controllers/search_controller.js

```diff
@@ -37,7 +37,7 @@ export default class extends Controller {
       content
         ? (this.searchedTarget.innerHTML = content.map((element) =>
             element.length > 2
-              ? `<a href=/posts/${element[0]}>${element[1]}`
+              ? `<a href=/posts/${element[0]}>${element[1]}</a>`
               : `<a href=/users/${element[0]}>${element[1]}</a>`
           ))
         : (this.searchedTarget.innerHTML = `<p>User or Post does not exist</p>`);
```

Comment:
Ternary operator must be clearly readable in just one line (you can break in two lines, positive evaluation and negative evaluation). Chaining ternary operator or executing large piece of code on it, is a bad practice. For nested conditions or large pieces of code, is better to use the if statement.

You can use an if statement for the first condition, and ternary operator as nested condition.

Reply:
Thanks for the observation